.github/workflows/test.yml

```
@@ -119,8 +117,23 @@ jobs:
         if: env.IS_NOT_FORK == 'true' && env.IS_DEFAULT_NPM == 'true'
         uses: "./.github/actions/check_clean"
 
+      - name: Check if monorepo version is a dev version
+        run: |-
+          version=`jq -r ".version" package.json | cut -d "-" -f 1`
+          echo "The version is $version"
+          is_dev_version=`(echo $version | grep -e "-dev") && echo "true" || echo "false"`
```

Currently the regex will match with `3.0-preview-develop-branch` for example.

How about this? 

```suggestion
          is_dev_version=`(echo $version | grep -e "\d-dev\b") && echo "true" || echo "false"`
```

So it'll match the following:
- echo '3.0-dev' | grep -e "\d-dev\b"
- echo '3.0-dev.0' | grep -e "\d-dev\b"

But not this:
echo '3.0-develop' | grep -e "\d-dev\b"